connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/network/HttpsConnection.kt

```diff
@@ -0,0 +1,46 @@
+package com.theoplayer.android.connector.uplynk.network
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runInterruptible
+import java.io.InputStreamReader
+import java.net.URL
+import javax.net.ssl.HttpsURLConnection
+
+internal const val CONNECT_TIMEOUT_IN_MS = 30000
+internal const val READ_TIMEOUT_IN_MS = 30000
+
+internal class HttpsConnection {
+    suspend fun get(urlString: String): String = runInterruptible(context = Dispatchers.IO) {
+            var result: String
+            var connection: HttpsURLConnection? = null
+            try {
+                connection = (URL(urlString).openConnection() as HttpsURLConnection)
+                    .apply {
+                        requestMethod = "GET"
+                        connectTimeout = CONNECT_TIMEOUT_IN_MS
+                        readTimeout = READ_TIMEOUT_IN_MS
+                        doInput = true
+                    }
+
+                connection.connect()
+
+                val responseCode: Int = connection.getResponseCode()
+                result = when (responseCode) {
+                    HttpsURLConnection.HTTP_OK ->
+                        connection
+                            .inputStream
+                            .bufferedReader()
+                            .use { it.readText() }
+                    else -> "Error: $responseCode"
+                }
+            } catch (e: Exception) {
+                result = e.message ?: ""
```

Comment:
We shouldn't return a `String` result in case of an error. Instead, we should `throw` an exception (or just let the current exception propagate) and handle it further up, e.g. by dispatching an `error` event on the player.

Reply:
It looked as a good idea to catch exceptions sooner but indeed when I look on it twice it looks awkward. I'll rework this part

Reply:
I would rather have the original HTTP exception, rather than a weird parser exception from us [trying to parse a random error message as JSON](https://github.com/THEOplayer/android-connector/blob/ccca3882f8965cca2f78906341927ecea32970ca/connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/network/UplynkApi.kt#L12).